### ISSUE TITLE
ffmpeg not working?

### ISSUE THREAD

**Issue by hml21619** [sim0n00ps/OF-DL#556]
nothing is downloading since update. what am i missing here? is it the ffmpeg exe the issue? where  can i download new fmpeg?
thanks 

![Screenshot (18)](https://github.com/user-attachments/assets/8d1e39db-d845-442a-a281-954c41bf090b)

**Comment by neonnskye** [comment]
Same issue here

**Comment by melithine** [comment]
> nothing is downloading since update. what am i missing here? is it the ffmpeg exe the issue? where can i download new fmpeg? thanks
> 
> ![Screenshot (18)](https://private-user-images.githubusercontent.com/180718626/365334774-8d1e39db-d845-442a-a281-954c41bf090b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU3MjE3NDYsIm5iZiI6MTcyNTcyMTQ0NiwicGF0aCI6Ii8xODA3MTg2MjYvMzY1MzM0Nzc0LThkMWUzOWRiLWQ4NDUtNDQyYS1hMjgxLTk1NGM0MWJmMDkwYi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkwN1QxNTA0MDZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05ZTEwMjFjYjU4NzhhMGVjNTVmZDE2NGQ2YWE3NGFhZmEwYzhkY2ZiMDY4YWI1YTdkMzY4MDQ3MWQzMDBiYzJjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.C_zNIhOhmYVUThNy2wRcDh8kWb157_TI57Sek6NjVfE)

What version of ffmpeg is that? Looks like it's too old to support what we need for DRM.

**Comment by Koujibear76** [comment]
Same issue. Reverted to 1.7.74 and its working fine now.

**Comment by bldgt92** [comment]
I was able to get past this error after upgrading my ffmpeg
https://github.com/sim0n00ps/OF-DL/issues/556#issue-2511407172

When I try to download a single paid message which is a DRM video, I get this error where it couldn't find the file. I am using the cdrm-project and I am not sure if that is the cause of this issue. This was all working for me a few weeks ago when I was on v1.7.71. 
![Annotation 2024-09-07 150413](https://github.com/user-attachments/assets/638dca91-adb0-4113-ac69-c778286636fb)

**Comment by hml21619** [comment]
> > nothing is downloading since update. what am i missing here? is it the ffmpeg exe the issue? where can i download new fmpeg? thanks
> > ![Screenshot (18)](https://private-user-images.githubusercontent.com/180718626/365334774-8d1e39db-d845-442a-a281-954c41bf090b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU3MjE3NDYsIm5iZiI6MTcyNTcyMTQ0NiwicGF0aCI6Ii8xODA3MTg2MjYvMzY1MzM0Nzc0LThkMWUzOWRiLWQ4NDUtNDQyYS1hMjgxLTk1NGM0MWJmMDkwYi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkwN1QxNTA0MDZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05ZTEwMjFjYjU4NzhhMGVjNTVmZDE2NGQ2YWE3NGFhZmEwYzhkY2ZiMDY4YWI1YTdkMzY4MDQ3MWQzMDBiYzJjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.C_zNIhOhmYVUThNy2wRcDh8kWb157_TI57Sek6NjVfE)
> 
> What version of ffmpeg is that? Looks like it's too old to support what we need for DRM.

i was using the ffmpeg that i got from the url of the orignal post. i think from ffmpeg.org
on this new program, it says the old ffmpeg doesnt work, so it auto downloaded some other ffmpeg file.
i think i can figure it out if i have a source to get the updated ffmpeg file.
or maybe i can just go back to using the old program like the others?
thank you

**Comment by wireman21** [comment]
Reverting back to 1.7.74 seems to work for me too. Thank you.

**Comment by hml21619** [comment]
> Reverting back to 1.7.74 seems to work for me too. Thank you.

hello, what version ffmpeg exe u use with 1.7.74?
thank you

**Comment by hml21619** [comment]
hey, you all can close this thread if u like. i downloaded a recent ffmpeg exe from ffmpeg . org
i also went back to 1.7.74. i also shut off my vpn ( maybe the VPN is the issue) is there a way to use this with a vpn?
thanks again

**Comment by wireman21** [comment]
> hey, you all can close this thread if u like. i downloaded a recent ffmpeg exe from ffmpeg . org i also went back to 1.7.74. i also shut off my vpn ( maybe the VPN is the issue) is there a way to use this with a vpn? thanks again

i am using vpn as OF is blocked in my country, vpn works for me.

**Comment by Leryn** [comment]
same, back to 1.7.74 and it seems ok. weird

**Comment by bldgt92** [comment]
> same, back to 1.7.74 and it seems ok. weird

Are you able to download DRM videos on 1.7.74? Do you use the cdrm-project or widevine keys? I use the cdrm project and it doesnt let me download any DRM videos. Regular videos and images download fine with 1.7.75.

**Comment by Leryn** [comment]
> > same, back to 1.7.74 and it seems ok. weird
> 
> Are you able to download DRM videos on 1.7.74? Do you use the cdrm-project or widevine keys? I use the cdrm project and it doesnt let me download any DRM videos. Regular videos and images download fine with 1.7.75.


yes i'm able to download drm videos and i use cdrm project, sometimes it still fails to download but i try again over and over the same link by "download single post" and it works

**Comment by melithine** [comment]
Please try v1.7.76: https://github.com/sim0n00ps/OF-DL/releases/tag/OFDLV1.7.76

**Comment by bldgt92** [comment]
> Please try v1.7.76: https://github.com/sim0n00ps/OF-DL/releases/tag/OFDLV1.7.76

Thanks, I tried with 1.7.76 and am still running into the same file path error as before, only for DRM videos.
![Annotation 2024-09-08 181045](https://github.com/user-attachments/assets/e1ceea6c-7a39-4670-887e-d97e4a60b672)

**Comment by bldgt92** [comment]
In case if you need the debug logs, i have attached those. I get the same error, no matter how many times I try for either posts or messages with DRM videos.
[SinglePost_Failure.txt](https://github.com/user-attachments/files/16924368/SinglePost_Failure.txt)

**Comment by melithine** [comment]
Cdrm project is down atm for maintenance. Hit me up on Discord for Widevine keys.

**Comment by bldgt92** [comment]
> Cdrm project is down atm for maintenance. Hit me up on Discord for Widevine keys.

I tried out with widevine keys and I get the exact same error as before. Something is messed up in the file path based on what I see in the logs.
```
"https://cdn3.onlyfans.com/dash/files/d/dd/dd7a1906edfd04332b7903695ed9c5e6/0htad3aoz3huervmrn6mf.mpd" -codec copy "H:\Videos\Backups\nwu\nw_of_drm\contentcreatorname/Posts/Free/Videos/0htad3aoz3huervmrn6mf_source.mp4"
2024-09-09 07:33:46.382 -07:00 [ERR] Exception caught: Could not find file 'H:\Videos\Backups\nwu\nw_of_drm\contentcreatorname\Posts\Free\Videos\0htad3aoz3huervmrn6mf_source.mp4'.
```
[SinglePost_Widevine_Updated.txt](https://github.com/user-attachments/files/16932414/SinglePost_Widevine_Updated.txt)

**Comment by hydr0homiee** [comment]
I was encountering this issue as well, what fixed it for me was downloading the newest version of FFmpeg and setting it as a PATH variable. Then adding the path to `FFmpeg.exe` in the OF-DL config.

**Comment by Koujibear76** [comment]
1) Download the latest version of the FFmpeg package from FFmpeg.org (version 7.0.2)
2) Move/Copy FFmpeg.exe to you OF-DL directory (all you need is FFmpeg.exe)
3) Make sure you use the latest version of OF-DL (version 1.7.76)

THAT'S IT. FIXED.

**Comment by bldgt92** [comment]
@hydr0homiee @Koujibear76 - Thank you very much. I think I had dowloaded the master build of ffmpeg instead of the release build. I downloaded and used 7.0.2 release build of ffmpeg and I am able to download the DRM videos now.

**Comment by hydr0homiee** [comment]
@melithine I think this is good to close 👍

**Comment by Leryn** [comment]
> I was encountering this issue as well, what fixed it for me was downloading the newest version of FFmpeg (7.0.2) and setting it as a PATH variable. Then adding the path to `FFmpeg.exe` in the OF-DL config.
> 
> OP you can download the newest version here: https://www.ffmpeg.org/download.html


sorry for my ignorance, what do you mean with path variable?

**Comment by hydr0homiee** [comment]
> > I was encountering this issue as well, what fixed it for me was downloading the newest version of FFmpeg (7.0.2) and setting it as a PATH variable. Then adding the path to `FFmpeg.exe` in the OF-DL config.
> > OP you can download the newest version here: https://www.ffmpeg.org/download.html
> 
> sorry for my ignorance, what do you mean with path variable?

Adding ffmpeg as a path variable lets you call ffmpeg from the Command Prompt.

https://www.wikihow.com/Install-FFmpeg-on-Windows

**Comment by Koujibear76** [comment]
> > > I was encountering this issue as well, what fixed it for me was downloading the newest version of FFmpeg (7.0.2) and setting it as a PATH variable. Then adding the path to `FFmpeg.exe` in the OF-DL config.
> > > OP you can download the newest version here: https://www.ffmpeg.org/download.html
> > 
> > 
> > sorry for my ignorance, what do you mean with path variable?
> 
> Adding ffmpeg as a path variable lets you call ffmpeg from the Command Prompt. It's covered towards the end of this article.
> 
> https://www.wikihow.com/Install-FFmpeg-on-Windows

I feel like it is just infinitely simpler to just copy ffmpeg.exe into your OF-DL folder rather than try to figure out PATH variables. Remember you can just do that and save yourself the headache. 👍🤓

**Comment by hydr0homiee** [comment]
I never said it isn't, but OF-DL isn't the only thing I use FFMpeg for and it's not like entering one command into your Command Prompt is all that difficult either lol.

**Comment by hydr0homiee** [comment]
> > > > I was encountering this issue as well, what fixed it for me was downloading the newest version of FFmpeg (7.0.2) and setting it as a PATH variable. Then adding the path to `FFmpeg.exe` in the OF-DL config.
> > > > OP you can download the newest version here: https://www.ffmpeg.org/download.html
> > > 
> > > 
> > > sorry for my ignorance, what do you mean with path variable?
> > 
> > 
> > Adding ffmpeg as a path variable lets you call ffmpeg from the Command Prompt. It's covered towards the end of this article.
> > https://www.wikihow.com/Install-FFmpeg-on-Windows
> 
> I feel like it is just infinitely simpler to just copy ffmpeg.exe into your OF-DL folder rather than try to figure out PATH variables. Remember you can just do that and save yourself the headache. 👍🤓

I never said it isn't, but OF-DL isn't the only thing I use FFMpeg for and it's not like entering one command into your Command Prompt is all that difficult either lol.

**Comment by Leryn** [comment]
sorry if i ask here an off topic question, but in this way i don't open a new post, is there a way to download only the videos you want making a list? i mean, i can download the single post, but every time i must copy and paste the link and it's very hard when i have hundreds of videos to download. it would be useful to have a way to download a list of specific links that i copied to download them automatically

**Comment by jasmeralia** [comment]
> sorry if i ask here an off topic question, but in this way i don't open a new post, is there a way to download only the videos you want making a list? i mean, i can download the single post, but every time i must copy and paste the link and it's very hard when i have hundreds of videos to download. it would be useful to have a way to download a list of specific links that i copied to download them automatically

Not currently, no. You'd have to file a feature request for that.

**Comment by Leryn** [comment]
> > sorry if i ask here an off topic question, but in this way i don't open a new post, is there a way to download only the videos you want making a list? i mean, i can download the single post, but every time i must copy and paste the link and it's very hard when i have hundreds of videos to download. it would be useful to have a way to download a list of specific links that i copied to download them automatically
> 
> Not currently, no. You'd have to file a feature request for that.


where? Pull requests section?

**Comment by melithine** [comment]
> > > sorry if i ask here an off topic question, but in this way i don't open a new post, is there a way to download only the videos you want making a list? i mean, i can download the single post, but every time i must copy and paste the link and it's very hard when i have hundreds of videos to download. it would be useful to have a way to download a list of specific links that i copied to download them automatically
> > 
> > 
> > Not currently, no. You'd have to file a feature request for that.
> 
> where? Pull requests section?

No, just another issue that can be labeled as an enhancement. Pull requests are appreciated if you can code it, of course.
